client/app/scripts/superdesk-authoring/authoring.js

```diff
@@ -871,11 +982,15 @@
                 $scope.$on('item:lock', function(_e, data) {
                     if ($scope.item._id === data.item && !_closing &&
                         session.sessionId !== data.lock_session) {
-                        var path = $location.path();
-                        if (path.indexOf('/view') < 0) {
-                            authoring.lock($scope.item, data.user);
+                        authoring.lock($scope.item, data.user);
+                        if ($scope.action !== 'view') {
                             $location.url($scope.referrerUrl);
                         }
+                        // var path = $location.path();
+                        // if (path.indexOf('/view') < 0) {
+                        //     authoring.lock($scope.item, data.user);
+                        //     $location.url($scope.referrerUrl);
+                        // }
```

Comment:
Same comment as above (there is a history in case we need the old code snippet back).